cmake/faceliftMacros.cmake

```diff
@@ -299,27 +300,32 @@ function(facelift_add_interface TARGET_NAME)
 
     endif()
 
+    if(ARGUMENT_MONOLITHIC_SUPPORTED)
+        set(ARGUMENT_MONOLITHIC MONOLITHIC_SUPPORTED)
+        message("OOOO ${LIBRARY_NAME}")
+    endif()
+
+    if(TARGET FaceliftIPCLibDBus)
+        set(PRIVATE_INCLUDE_DIRECTORIES ${IPC_DBUS_OUTPUT_PATH})
+    endif()
+
     facelift_add_library(${LIBRARY_NAME}
         SOURCES_GLOB_RECURSE ${MODULE_OUTPUT_PATH}/*.cpp
         HEADERS_GLOB_RECURSE ${MODULE_OUTPUT_PATH}/*.h
         LINK_LIBRARIES FaceliftModelLib ${ARGUMENT_LINK_LIBRARIES}
+        MONOLITHIC_LINK_LIBRARIES ${ARGUMENT_MONOLITHIC_LINK_LIBRARIES}
         PUBLIC_HEADER_BASE_PATH ${MODULE_OUTPUT_PATH}
         ${ARGS}
         UNITY_BUILD
+        PRIVATE_DEFINITIONS
+            ${MODULE_COMPILE_DEFINITIONS}
+        ${ARGUMENT_MONOLITHIC}
+        PRIVATE_INCLUDE_DIRECTORIES ${PRIVATE_INCLUDE_DIRECTORIES}
     )
 
-    if(TARGET FaceliftIPCLibDBus)
-        target_include_directories(${LIBRARY_NAME}
-            PRIVATE
-                $<BUILD_INTERFACE:${IPC_DBUS_OUTPUT_PATH}>
-        )
-    endif()
-
-    target_compile_definitions(${LIBRARY_NAME} PRIVATE "${MODULE_COMPILE_DEFINITIONS}")
-
     # Get the list of files from the interface definition folder so that we can regenerate the code whenever there is a change there
     file(GLOB_RECURSE QFACE_FILES ${ARGUMENT_INTERFACE_DEFINITION_FOLDER}/*.qface)
-    target_sources(${LIBRARY_NAME} PRIVATE ${QFACE_FILES})
+#    target_sources(${LIBRARY_NAME} PRIVATE INTERFACE ${QFACE_FILES})
```

Comment:
Please don't leave commented out lines. Remove it or add reason why it is commented out.